### ISSUE TITLE
No backwards compatible way to specify `since` and `deprecated` without warnings

### ISSUE THREAD

**Issue by kipcole9** [elixir-lang/elixir#7937]
### Environment

* Elixir 1.7.0-rc.1 (b6d776967) (compiled with Erlang/OTP 20)

### Current behavior

Code using `@since` and `@deprecated` issues a compiler warning:

```elixir
warning: module attribute @since is deprecated, please use "@doc since: ..." instead
  lib/money.ex:341
```

Yet using the new `@doc since: "1.2.3"` that silences warnings in `1.7.0-rc.*` will raise compilation errors in current and older versions of Elixir.

Therefore there is no backwards compatible way to use `since` and `deprecated` without conditional compilation.

As a lib writer I strive to have zero compilation warnings since I think that warnings, to a lib consumer, convey at minimum carelessness and may potentially deter adoption or usage.

### Expected behavior

Don't emit deprecation warning for `@since` and `@deprecated` in `1.7.*`

**Comment by josevalim** [comment]
@kipcole9 `@since` was never ever used, so adding it has no effect whatsoever. `@deprecated` has not been deprecated though, so you should continue using it. Does it make sense?

TL;DR - remove `@since` as it has no effect and add `@doc since: ...` once you are on v1.7+. Keep `@deprecated as is`.

**Comment by kipcole9** [comment]
Hmmm.  `@since` is at least a handy doc item but your point is understood.

The challenge is that as soon as I add `@doc since: "1.2.3"` I can no longer compiler on older releases of Elixir.  The current code compiles and runs back to Elixir 1.5.  Which means I would have to create a release that only runs on versions later than `1.7.0` or do conditional compilation which would be very smelling for a meta data format change, no?

**Comment by kipcole9** [comment]
Got it, thanks José.  Closing.

**Comment by OvermindDL1** [comment]
You can also generate the @doc code via a Mix version test or so
